service/src/main/java/org/databiosphere/workspacedataservice/dao/PostgresInstanceDao.java

```
@@ -14,6 +18,9 @@ public class PostgresInstanceDao implements InstanceDao {
 
   private final NamedParameterJdbcTemplate namedTemplate;
 
+  @Value("${twds.instance.workspace-id}")
+  private UUID workspaceId;
```

I think to support the notion that WDS should fail fast if there's anything fishy with `WORKSPACE_ID` the creation of the `PostgresInstanceDao` bean should be `DeploymentMode` aware and strongly enforce the non-null `workspaceId` assumption, perhaps in both directions.

Maybe sibling bean methods somewhere that roughly do the following:

```
@Value("${twds.instance.workspace-id}")
private UUID workspaceId;

@ControlPlane
InstanceDao instanceDao instanceDao() {
    assert(workspaceId == null, "workspaceId must NOT be set in control plane");
    return // new instance of PosgresInstanceDao using the existing single-arg constructor
}

@DataPlane
InstanceDao instanceDao instanceDao() {
    assertLegitWorkspaceId(workspaceId, "workspaceId must legit in the data plane");
    return // new instance of PosgresInstanceDao using an overloaded two-arg constructor with a `@NonNull WorkspaceId` param
}
```

---

(or, thinking further on this)...maybe the `InstanceDaoProvider` itself should be the bean, and you get one instance or the other based on the DeploymentMode, and the provider either has (or doesn't have) a `WorkspaceId`?

---

Regardless of _how_, I just want to stress the main point which is that it'd be good to fail as early as possible if our assumptions around `WORKSPACE_ID` don't match up with the deploymentMode.

---

As of #509 , all this validation has moved to `StartupConfig`, which should handle validation for InstanceDao as well as any other classes that rely on the `WORKSPACE_ID` value